src/javascript/gameInstance.js

```diff
@@ -120,12 +124,15 @@ export default class GameInstance {
     this.endX = startX + (trackWidth + 1) * this.dropTrackArr.length - 1;
 
     const isMobile = window.innerWidth < 1000;
-    const hitLineProp = isMobile ? 8.5 : 9;
+    let hitLineProp = isMobile ? 8.5 : 9;
+    if (!this.vm.playMode) hitLineProp = this.vm.options.lowerHitLine ? 4 : 0;
 
-    this.vm.checkHitLineY = (this.canvas.height / 10) * hitLineProp;
-    this.vm.noteSpeedPxPerSec =
-      this.vm.checkHitLineY / Number(this.vm.noteSpeedInSec);
+    this.checkHitLineY = (this.canvas.height / 10) * hitLineProp;
+    this.noteSpeedPxPerSec = 380 * Number(this.vm.noteSpeed);
+    this.noteDelay = this.checkHitLineY / this.noteSpeedPxPerSec;
+    await this.getCurrentTime();
```

Comment:
This method name suggests that `this.getCurrentTime()` returns some value, i.e. it should return some `Promise<Time>`. If this method is actually mutating some field on `this`, I'd recommend renaming it to `this.updateCurrentTime()` or something.

Reply:
Yeah but this function does return value and it is been used somewhere else. Here it is only to update the value. Maybe I could try to make this clearer somehow later

Reply:
That seems like a bad idea to update AND get, cuz now you're mixing getting which is pure, and updating with side-effects?

Reply:
Right, will look into this

Reply:
Yes, I know this is a bit weird, so I have looked at in what places this function is being invoked, so basically we have a `gameTimingLoop` interval runs every 100 ms, to update on note times check if game ends etc, and the function is called there each time to update the time in itself, return value is not used. And apart from this, the function is called on users input, to get the most accurate time (other than every 100 ms) when the user hits or creates the note.

I'm not sure why I have chosen 100 ms to be exact, will try solve this.